File: LC_JS-SOLUTIONS/121. Best Time to Buy and Sell Stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
  if (prices.length === 0) {
    return 0;
  }

  let maxProfit = 0;
  let minProfit = prices[0];

  for (let price of prices) {
    minProfit = Math.min(minProfit, price);
    profit = price - minProfit;
    maxProfit = Math.max(profit, maxProfit);
  }

  return maxProfit;
};

File: LC-Python-Solution/121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        if not prices:
            return 0
        
        maxProfit = 0
        minPrice = prices[0]

        for price in prices:
            minPrice = min(price,minPrice)
            profit = price - minPrice
            maxProfit = max(maxProfit , profit)
        
        return maxProfit



        